shadowsocks/udp_buffer_pool.go

```diff
@@ -0,0 +1,22 @@
+package shadowsocks
+
+import "sync"
+
+// maxUDPBufferSize is the maximum UDP packet size in bytes.
+const maxUDPBufferSize = 16 * 1024
+
+var pool = sync.Pool{
```

Comment:
I wonder if `Pool` is a net win for us.  Using `Pool` reduces the number of allocations, but it probably increases the peak non-collectible memory usage, because the buffers that are in use at any given time will be oversized for the data they need to hold.  Also, it seems like recently used buffers in the pool may be temporarily immune from garbage collection, which would further increase peak memory usage.